src/main/java/uk/gov/hmcts/ccd/auditlog/AuditEntry.java

```diff
@@ -0,0 +1,169 @@
+package uk.gov.hmcts.ccd.auditlog;
+
+import org.apache.commons.lang3.builder.ReflectionToStringBuilder;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+import static org.apache.commons.lang3.builder.ToStringStyle.JSON_STYLE;
+
+public class AuditEntry {
```

Comment:
with this design it's not possible from the code to distinguish what properties we are logging for each operation type. 
I wonder whether it's possible to have an AuditEntry superclass with one implementation for each operation type that contains properties only relevant to that operation. So for example CaseSearchAudit, CaseUpdateAudit....

Reply:
you get that from LogAudit() annotation on the controller method.  I feel defining another class hierarchy complicates aspect design.